specification/maps/data-plane/Creator/preview/1.0/spatial.json

```diff
@@ -25,15 +25,21 @@
       "scopes": {
         "https://atlas.microsoft.com/.default": "https://atlas.microsoft.com/.default"
       }
     },
     "AzureKey": {
       "type": "apiKey",
-      "description": "This is a shared key that is provisioned when creating an [Azure Maps resource](https://aka.ms/amauth) through the Azure management plane  via Azure portal, PowerShell, CLI, Azure SDKs, or REST APIs.\n\n With this key, any application is authorized to access  all REST APIs. In other words, these can currently be treated as master keys to the account which they are issued for.\n\n For publicly exposed applications, our recommendation is to use server-to-server access of Azure Maps REST APIs where this key can be  securely stored.",
+      "description": "This is a shared key that is provisioned when creating an [Azure Maps resource](https://aka.ms/amauth) through the Azure management plane  via Azure portal, PowerShell, CLI, Azure SDKs, or REST APIs.\n\n With this key, any application is authorized to access  all REST APIs. In other words, these can currently be treated as master keys to the account which they are issued for.\n\n For publicly exposed applications, our recommendation is to use server-to-server access of Azure Maps REST APIs where this key can be securely stored.",
       "name": "subscription-key",
+      "in": "query"
+    },
+    "SasToken": {
+      "type": "apiKey",
```

Comment:
@stack111: SaSToken use oauth2 bearer tokens right? If it is so, even if this custom auth definition is ignored by autorest, wouldn't it better to specify "type": "oauth2" to clarify the intention?

Reply:
SasToken is a JWT bearer token but it is not acquired through oauth2 protocol. It's only acquired through ARM APIs and doesn't match any RFC. Intentionally I did not want to leave an impression that someone could use oauth2 to acquire the sas token.